src/main/java/io/cryostat/discovery/DiscoveryStorage.java

```diff
@@ -324,6 +356,11 @@ public List<? extends AbstractNode> update(
 
     public List<? extends AbstractNode> update(
             UUID id, Collection<? extends AbstractNode> children, boolean notify) {
+        // FIXME what about required SecurityContext fields on TargetNodes? We find the JVM ID here
```

Comment:
Plugin registration currently has no notion of Namespaces. In the comment here I think I was trying to imply that the discovery plugin registration API should allow plugins to include some Security Context information that can be hooked into the system from this PR, but I'm not fully sure yet what exactly that will look like. It should be a generic property that can vary with different plugins and different Cryostat deployments scenarios, ie it should not enforce something like a Namespace value that may not be present or applicable to every platform.

Reply:
Oh, and plugins are not necessarily discoverable targets themselves - they don't even need to be Java applications. So we can't assume we can do something like correlate the node's realm to a plugin and then assume that the realm is provided by some target we see and can understand its namespace from.

Also, even if the plugin is in fact a JVM application and it is a discoverable target, there is no restriction that says that the plugin is only discovering targets within its own namespace. The plugin could be privileged and be able to see targets in many namespaces or across the whole cluster and publish that information to Cryostat.

Reply:
So a plugin does not have any notion of namespaces, but they do have to be a descendant resource of a namespace  in the OpenShift deployment tree scheme right?

Reply:
In an OpenShift context a discovery plugin probably is also a container running as part of some deployment within some namespace, yes. But the plugin isn't necessarily a Java application, or exposing a JMX connection to itself, or otherwise present as part of the tree that Cryostat is aware of.

Reply:
And in practical terms of why I don't want to add that restriction (forcing plugins to be JVM-mode applications with a JMX port open), plugins should be very small and lightweight on their own so there are probably better runtimes to build plugins with. Quarkus native image mode with the Java operator-sdk, or Go with operator-sdk, for example. The former should eventually be connectable over JMX and could be both a plugin as well as a target, but the Go plugin never would be.

Also, the Cryostat Agent in push-mode would probably be attached to a target application that does not accept JMX connections, but the Agent would still publish its presence to Cryostat so that Cryostat can use that information to collect the pushed data and store all the pushed files as belonging to the same target.